### PR TITLE
[AV1e] allow user select entrypoint for av1 encoder

### PR DESCRIPTION
**Pull request by leyu-yao**
This PR add support to allow user select entrypoint between VAEntrypointEncSlice and VAEntrypointEncSliceLP.

### CODE REVIEW

File: encode/av1encode.c [hunk 1]
```diff
@@ -875,7 +887,43 @@ static int init_va(void)
...
+            entryPoint = entrypoints[slice_entrypoint];
+            break;
+        }
+    }
+
+    if(entrypoints)
+        free(entrypoints);
+
+    if (support_encode == 0) {
+        printf("Can't find avaiable or request entrypoints for AV1 profiles\n");
```
Comment (yellowdoge) [inline encode/av1encode.c:924]: s/avaiable or request/available or requested/

File: encode/av1encode.c [hunk 2]
```diff
@@ -875,7 +887,43 @@ static int init_va(void)
...
+
+    int num_entrypoints = vaMaxNumEntrypoints(va_dpy);
+    VAEntrypoint* entrypoints = malloc(num_entrypoints * sizeof(*entrypoints));
+    if (!entrypoints) {
+        fprintf(stderr, "error: failed to initialize VA entrypoints array\n");
+        exit(1);
+    }
+
+    vaQueryConfigEntrypoints(va_dpy, av1_profile, entrypoints, &num_entrypoints);
+    int support_encode = 0;
```
Comment (yellowdoge) [inline encode/av1encode.c:901]: Can be bool?

**Review by yellowdoge — APPROVED** [review]
Non owner LGTM!

File: encode/av1encode.c (hunk 2 above)
Comment (leyu-yao) [reply encode/av1encode.c:901]: It can be bool. But I will just follow the coding style with other codecs.

File: encode/av1encode.c (hunk 1 above)
Comment (leyu-yao) [reply encode/av1encode.c:924]: Will update it. Thanks for catching this.

**Review by XinfengZhang — APPROVED** [review]

**Comment by lobomela** [comment]
Can we get this merged ?

**Pull request merged**